python/tank/pipelineconfig.py

```diff
@@ -244,7 +244,66 @@ def __repr__(self):
 
     ########################################################################################
     # handling pipeline config metadata
-    
+
+    def get_roots_metadata_location(self):
+        """
+        Returns the location to the roots metadata.
+        """
+        return os.path.join(
+            self._pc_root, "config", "core", constants.STORAGE_ROOTS_FILE
+        )
+
+    def _get_roots_metadata(self):
+        """
+        Loads and validates the roots metadata file.
+
+        The roots.yml file is a reflection of the local storages setup in Shotgun
+        at project setup time and may contain anomalies in the path layout structure.
+
+        The roots data will be prepended to paths and used for comparison so it is
+        critical that the paths are on a correct normalized form once they have been
+        loaded into the system.
+
+        :returns: A dictionary structure with an entry for each storage defined. Each
+                  storage will have three keys mac_path, windows_path and linux_path,
+                  for example
+                  { "primary"  : <ShotgunPath>,
+                    "textures" : <ShotgunPath>
+                  }
+        """
+        # now read in the roots.yml file
+        # this will contain something like
+        # {'primary': {'mac_path': '/studio', 'windows_path': None, 'linux_path': '/studio'}}
+        roots_yml = self.get_roots_metadata_location()
+
+        try:
+            # if file is empty, initialize with empty dict...
+            data = yaml_cache.g_yaml_cache.get(roots_yml, deepcopy_data=False) or {}
+        except Exception as e:
+            raise TankError("Looks like the roots file is corrupt. Please contact "
+                            "support! File: '%s' Error: %s" % (roots_yml, e))
+
+        # if there are more than zero storages defined, ensure one of them is the primary storage
+        if len(data) > 0 and constants.PRIMARY_STORAGE_NAME not in data:
+            raise TankError("Could not find a primary storage in roots file "
+                            "for configuration %s!" % self._pc_root)
+
+        # sanitize path data by passing it through the ShotgunPath
+        shotgun_paths = {}
+        for storage_name in data:
+            shotgun_paths[storage_name] = ShotgunPath.from_shotgun_dict(data[storage_name])
+
+        return shotgun_paths
+
+    def _get_config_core_folder(self):
```

Comment:
Do we need this? It just adds complexity for me with no apparent benefit. I find the code more readable if we do for example
```
def get_core_hooks_location():
  return os.path.join(self.get_config_location(), "core", "hooks")
```

Fewer levels of indirection.

Reply:
Removed.